src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -25,10 +25,11 @@
     @Mock
     private RequestDispatcher requestDispatcher;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    private final ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
     public void setup(){
+        servlet.init();
```

Comment:
Everything except the tested class should be mocked. In that case, product dao should be a @<!-- -->Mock field, injected into the tested instance via @<!-- -->InjectMocks, that also returns something meaningful (e.g. a singleton list of one @<!-- -->Mock product) from its findAll() method